cabby/model/landmark_recognition/dataset_bert.py

```diff
@@ -0,0 +1,180 @@
+  train_dataset = EntityRecognitionSplit(
+    rvs_dataset.train)
+  logging.info(
+    f"Finished to create the train-set with {len(train_dataset)} samples")
+  val_dataset = EntityRecognitionSplit(
+    rvs_dataset.valid)
+  logging.info(
+    f"Finished to create the valid-set with {len(val_dataset)} samples")
+  test_dataset = EntityRecognitionSplit(
+    rvs_dataset.test)
+  logging.info(
+    f"Finished to create the test-set with {len(test_dataset)} samples")
+
+  return train_dataset, val_dataset, test_dataset
+
+def basic_tokenize_and_align_labels(
+  sentence: str, text_labels: Dict[str,Tuple[int, int]]
+) :
+  '''Tokenize sentence and preserve the labels,
+  such that for each token there is a label.
+  Arguments:
+    sentence: the sentence to tokenize.
+    text_labels: the dictionary containing the landmarks (keys) and span (values).
+  Returns:
+    A List of the tokens and their corresponding labels.
+  '''
+
+  # Basic tokenization of the sentence.
+  sentence_words = word_tokenize(sentence)
+
+  # Create labels for each token according to the dictionary of entities and spans.
+  labels = []
+  spans = sorted(list(text_labels.values()))
+
+  cur_index = 0
+  span = spans.pop(0)
+  start, end = span[0], span[1]
+  for word in sentence_words:
+    if cur_index >= start:
+      labels.append(1)
+    else:
+      labels.append(0)
+
+    cur_index += len(word)
+    if cur_index<len(sentence) and sentence[cur_index] == ' ':
+      cur_index+=1
+    if cur_index >= end:
+      if len(spans)>0:
+        span = spans.pop(0)
+        start, end = span[0], span[1]
+      else:
+        start = len(sentence)+1
+
+  return sentence_words, labels
+
+def bert_tokenize_and_align_labels(
+  tokenized_sentence: List[int], tags: List[int]
+) -> Dict[str, torch.Tensor]:
+  '''Bert-tokenization of sentence and preserve the labels,
+  such that for each token there is a label.
+  Arguments:
+    tokenized_sentence: the tokenized sentence.
+    tags: the labels corresponding to the tokens.
+  Returns:
+    A dictionary of Bert-tokenized sentences and corresponding labels.
+  '''
+  tokenized_inputs = tokenizer(tokenized_sentence, truncation=True, is_split_into_words=True)
+  word_ids = tokenized_inputs.word_ids(batch_index=0)
+  previous_word_idx = None
+  label_ids = []
+  for word_idx in word_ids:
+    # Special tokens have a word id that is None.
+    # We set the label to -100 so they are automatically
+    # ignored in the loss function.
+    if word_idx is None:
+      label_ids.append(-100)
+    # We set the label for the first token of each word.
+    elif word_idx != previous_word_idx:
+      label_ids.append(tags[word_idx])
+    # For the other tokens in a word, we set the label to the current label,
+    else:
+      label_ids.append(tags[word_idx])
+    previous_word_idx = word_idx
+
+  assert len(label_ids) == len(tokenized_inputs['input_ids'])
+
+  tokenized_inputs["labels"] = label_ids
+  return tokenized_inputs
+
+
+class PadSequence:

```

Comment:
Actually, there is even more (apparently) duplicated with dataset.py. Can you refactor so there isn't this duplication?

Reply:
I will remove dataset.py